Routers/Routers/Graph.cs

```diff
@@ -0,0 +1,127 @@
+namespace Routers;
+
+using Routers.Exceptions;
+
+/// <summary>
+/// Class realizes undirected graph abstraction.
+/// </summary>
+public class Graph
+{
+    private int componetsCount;
+
+    private UnionFind conectivityCheck = new ();
+
+    /// <summary>
+    /// Gets the number of verticies in the graph.
+    /// </summary>
+    public int Size { get => this.AdjacencyLists.Count; }
+
+    /// <summary>
+    /// Gets a value indicating whether the graph is connected or not.
+    /// </summary>
+    public bool IsConnected { get => this.componetsCount == 1; }
```

Comment:
```suggestion
    public bool IsConnected => this.componetsCount == 1;
```
И выше тоже можно упростить